rfc/rfc-0000-move-off-monorepo.md

```diff
@@ -0,0 +1,57 @@
+- as we continue to build up the Cartographer ecosystem with integrations, adaptors, platform controllers this will cause a bottleneck for changes proposed on a single repo.
+
+Adopting a microservice approach for Cartographer will aid extensions and help build up the wider ecosystem and community around it.
+This RFC proposes how we will adopt this microservice style approach that will help grow the Cartographer ecosystem, encourage new integrations and extensions is a sustainable and maintainable way that suits the project and its aspirations.
+
+## Motivation
+
+The motivation behind Cartographer as an OSS project aims to build a community and ecosystem to aid supply chain choreography.  Cartographer itself is an "umbrella" for this ecosystem, the first step of this was to build the core components and now we are looking ahead to extend by adding more capabilities.
+ 
+As expressed by maintainers of the https://github.com/vmware-tanzu/cartographer repository it will become hard to manage it if we use it as a monorepo and continue to add more capabilities like conventions etc.
+It is expected that this RFC describes how we split the current repository so it allows the Cartographer OSS project to evolve and scale beyond the concrete capabilities the core aims to provide.
+For example splitting out from the current repo:
+
+-  the __`Runnable`__ feature so it can release independently and install standalone in a target deployment cluster,  avoids deploying all of Cartographer which contains build cluster CRDs not desired in runtime clusters
+- the __examples directory__ so we can build up a wide range of examples that include integrations with wider platform services like conventions.  This also helps encourage contribution of community supply chains for different integrations where we can add automated testing that runs against different setups.
+- the __website__ so we can add documentation for these new integrations and platform capabilities outside of the core repository.  I.e. Documenting services like the conventions or a Jenkins integration should not require a change to the core cartographer repository, this should live outside in its own repository.
+
+## Possible Solutions
+
+There seems to be only two options possible solutions:
+- __monorepo__ we could revisit the [convention proposal](https://github.com/vmware-tanzu/cartographer/pull/514) decision and look at ways to address the monorepo concerns however there seemed a general consensus on the [office hours call](https://www.youtube.com/watch?v=iCURQqV52Uw&t=1173s) that this was not desired.
+- __microservice__ As conventions are now being worked on in a different repository, we should look at splitting out parts that currently reside in https://github.com/vmware-tanzu/cartographer.
+
+Suggestions:
+
+-  __`Runnable`__ feature so it can install standalone in a target deployment cluster and avoids deploying all of Cartographer
```

Comment:
Not sure this is a high priority.

1. deploying runnable does mean a fatter controller and unnecessary CRDs today, but that can be mitigated in other ways if required.
2. runnable co-opts a lot of the cartographer runtime, so we'd have to libify cartographer as part of this effort.

I feel like adressing extracting runnable as it's own controller (or the same controller with feature flags to only run where the Runnable CRD is installed) is an orthoganal concern, but this RFC opens up our options for such future discussions.

Reply:
Would it be that much effort to split out runnable? In terms of the underlying code that's shared and called from the reconciler, it's all libified already, no? The runnable repo can still import `github.com/vmware-tanzu/cartographer/pkg/...` and I would think not much would change.

Reply:
In code, no. In docs? maybe, in tooling, plenty? The argument isn't about the size and shape of the work, the argument is the need to even consider it at this point. Perhaps I'm being too "agile" in wanting to focus on immediate pain points, of which there are enough to accept work to split out docs and be prepared for creating new sub-projects.

Reply:
Perhaps we should keep this RFC focussed on what our desired end state is and keep prioritisation a separate discussion?  That would help understand what's involved and encourage community contributions if folks felt it important to them?